### PR TITLE
initial creation of SlackBot Demo

### PR DESCRIPTION
**Pull request by sandyhider**
I wanted you all to look at the SlackBot API and see if you want me to any anything.  Right now you can get and set the dorset Application.

### CODE REVIEW

File: slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        slackBot.connectToSlack(apiToken);
+        Collection<String> channels = slackBot.getChannels();
+        System.out.println("Channels listing:");
+        for(String ch : channels) {
+        	System.out.println(ch);
+        }
+        System.out.println();
+        
+        slackBot.startListening();
+        System.out.println("Listening on bottesting channel in KossiTesting.slack.com.");
```
Comment (cash) [inline slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java]: Can you get the channel name from the API?

File: slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBot.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    private Application application = null;
+
+	private SlackMessagePostedListener messagePostedListener = null;
+    
+    
+    /**
+     * @param application Application for Dorset-Framework
+     */
+    public SlackBot(Application application) {
+        super();
```
Comment (cash) [inline slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBot.java]: Why super()?

File: slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import edu.jhuapl.dorset.agents.Agent;
+import edu.jhuapl.dorset.agents.EchoAgent;
+import edu.jhuapl.dorset.routing.Router;
+import edu.jhuapl.dorset.routing.SingleAgentRouter;
+
+public class SlackBotDemo {
+	
+	public String getSlackApiToken() {
+		Properties properties = new Properties();
+		InputStream in = SlackBot.class.getResourceAsStream("/slackbot.properties");
```
Comment (cash) [inline slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java]: Maybe use the try-with-resources statement to clean this up a bit.

File: slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+     */
+    public static void main(String[] args) throws IOException {
+    	
+
+        Agent agent = new EchoAgent();
+        Router router = new SingleAgentRouter(agent);
+        Application app = new Application(router);
+        SlackBotDemo demo = new SlackBotDemo();
+
+        System.out.println("Weclome to SlackBot Echo Test.");
```
Comment (cash) [inline slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java]: "Welcome"

**Comment by cash** [comment]
@sandyhider can you format the code in Eclipse using the code style from here: https://github.com/DorsetProject/dorset-framework/blob/master/.developers/eclipse_code_style.xml

I noticed some tabs and some spacing variations from our code style.

File: slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java (hunk 1 above)
Comment (sandyhider) [reply slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java]: Do you mean when I receive a message?  Yes.  I use the channel name when I send the message back.  Do you want me to pass the channel name into the application?

Comment (cash) [reply slackbot/src/main/java/edu/jhuapl/dorset/demos/SlackBotDemo.java]: I don't think we should have a hard coded channel name in the println. Is there way to replace that or is it unknown when it runs?

File: slackbot/pom.xml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+						<phase>package</phase>
+						<goals>
+							<goal>shade</goal>
+						</goals>
+						<configuration>
+							<shadedArtifactAttached>false</shadedArtifactAttached>
+							<transformers>
+								<transformer
+									implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+									<mainClass>edu.jhuapl.dorset.demos.SlackBot</mainClass>
```
Comment (cash) [inline slackbot/pom.xml]: Shouldn't this be SlackBotDemo rather than SlackBot?

**Pull request merged**